### DISCUSSION TITLE
Ignore or get rid of CssSyntaxError: <css input>: '<key>' does not exist in your theme config in a monorepo setup

### DISCUSSION THREAD

**Discussion by myang5** [tailwindlabs/tailwindcss#10510] [Help] [answered]
The codebase I'm working on has 3 different apps, plus a `core` folder with misc. code shared between the three apps. I have Tailwind set up so that each app has its own config, with some theme overrides specific to each app. For example, `app1` has the following `tailwind.config.js`:
```js
module.exports = {
  presets: [require('../../core/tailwind.config')],
  purge: ['./src/**/*.{js,jsx}', '../../core/lib/**/*.{js,jsx}'],
  theme: {
    extend: {
      spacing: {
        'app-header-height': '88px',
      },
    },
  },
};
```
As you can see from the config, I also have JIT mode set up so that each app's build process purges the respective app + the `core` folder, resulting in one `tailwind.css` file for each app. This is because having two files, one for the app and one for core, was causing unexpected behavior.

I have a component in the `core` folder with this style: `h-[calc(100vh-theme(spacing.app-header-height))]`
The component is only used in 2/3 of the apps, so the `app-header-height` value is only defined in 2/3 of the Tailwind configs. When I attempt to build the third app without `app-header-height` defined, I get this error:
```
CssSyntaxError: <css input>: 'spacing.app-header-height' does not exist in your theme config.
```

I know this is happening because each app's Tailwind config also purges the entire `core` folder, so the component that is used in 2/3 of the apps is getting included in the purge process of the third app. However, I'm not sure how to fix this error without some major restructuring of our code organization, or not taking advantage of the ability to access `theme` in custom values. Is there a clean workaround for this, or some config option I'm missing?

I can fix this quickly by defining a dummy value for `app-header-height` in the third app's Tailwind config, but that doesn't seem like a sustainable solution.

**Comment by wongjn** [comment, marked as answer]
If you use `app-header-height` in `core`, then it would seem sensible to define it in Tailwind config in `core` too, even if it was some dummy value. The apps can keep the config as they are to override it.

**Reply by myang5** [reply]
Hm ok, as long as the dummy value is guaranteed to be overwritten that's an improvement. I saw this in the presets documentation:
```
The exception to this is the extend key, which is collected across all configurations and applied on top of the rest of the theme configuration.
```
but I wasn't sure how what this meant for merging different `extends` keys.

I still feel like it's not ideal to have to include dummy values at all, but this may be more of an issue with our code structure.

**Reply by wongjn** [reply]
> I wasn't sure how what this meant for merging different `extends` keys.

If there are values with the same keys across a preset chain, then the "last" value would be the one that is used.

> I still feel like it's not ideal to have to include dummy values at all, but this may be more of an issue with our code structure.

It feels like there would have been some subconscious thought about the fallback value of `spacing.app-header-height` – it would *need* to exist in some form, as `calc(100vh - )` would not be a valid value for the `height` property if it did not exist.

**Reply by myang5** [reply]
That's fair, the Tailwind engine doesn't have specific knowledge of our monorepo structure so when the third app is building, it just sees that `app-header-height` is not defined, although the component is not used in the third app at all.
